src/engine/include/coincenteroptions.hpp

```diff
@@ -49,29 +44,110 @@ struct CoincenterCmdLineOptions {
   std::string withdraw{};
 };
 
-inline CoincenterCmdLineOptions ParseCoincenterCmdLineOptions(const vector<std::string_view>& vargv) {
-  auto parser = CmdOpts<CoincenterCmdLineOptions>::Create(
-      {{"--help", &CoincenterCmdLineOptions::help},
-       {"--version", &CoincenterCmdLineOptions::version},
-       {"--loglevel", &CoincenterCmdLineOptions::logLevel},
-       {"--logfile", &CoincenterCmdLineOptions::logFile},
-
-       {"--orderbook", &CoincenterCmdLineOptions::orderbook},
-       {"--orderbook-depth", &CoincenterCmdLineOptions::orderbook_depth},
-       {"--orderbook-cur", &CoincenterCmdLineOptions::orderbook_cur},
-
-       {"--balance", &CoincenterCmdLineOptions::balance},
-       {"--balance-cur", &CoincenterCmdLineOptions::balance_cur},
-
-       {"--trade", &CoincenterCmdLineOptions::trade},
-       {"--trade-strategy", &CoincenterCmdLineOptions::trade_strategy},
-       {"--trade-timeout", &CoincenterCmdLineOptions::trade_timeout_s},
-       {"--trade-emergency", &CoincenterCmdLineOptions::trade_emergency_ms},
-       {"--trade-updateprice", &CoincenterCmdLineOptions::trade_updateprice_ms},
-       {"--trade-sim", &CoincenterCmdLineOptions::trade_sim},
-
-       {"--withdraw", &CoincenterCmdLineOptions::withdraw}});
-  return parser.parse(vargv);
+template <class OptValueType>
+inline CommandLineOptionsParser<OptValueType> CreateCoincenterCommandLineOptionsParser() {
+  constexpr int64_t defaultTradeTimeout =
+      std::chrono::duration_cast<std::chrono::seconds>(api::TradeOptions::kDefaultTradeDuration).count();
+  constexpr int64_t emergencyBufferTime =
+      std::chrono::duration_cast<std::chrono::milliseconds>(api::TradeOptions::kDefaultEmergencyTime).count();
+  constexpr int64_t minUpdatePriceTime =
+      std::chrono::duration_cast<std::chrono::milliseconds>(api::TradeOptions::kDefaultMinTimeBetweenPriceUpdates)
+          .count();
+  const bool isSimulationModeByDefault = api::TradeOptions().simulation();
+
+  return CommandLineOptionsParser<OptValueType>(
+      {{{{"General", 1}, "--help", 'h', "", "Display this information"}, &OptValueType::help},
```

Comment:
We could try to beautify a bit these in the future, by managing manually the indentation.

Reply:
yeah, I think we will add a :
// clang-format off